ReactWindows/ReactNative.Shared/Views/Control/ReactControlManager.cs

```diff
@@ -0,0 +1,440 @@
+using Newtonsoft.Json.Linq;
+using ReactNative.Reflection;
+using ReactNative.UIManager;
+using ReactNative.UIManager.Annotations;
+using ReactNative.UIManager.Events;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+#if WINDOWS_UWP
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Input;
+#else
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Input;
+using System.Windows.Media;
+#endif
+
+namespace ReactNative.Views.ControlView
+{
+    /// <summary>
+    /// View manager for React control instances.
+    /// </summary>
+    public class ReactControlManager : ViewParentManager<ReactControl, ReactControlShadowNode>
+    {
+        private const int FocusCommand = 1;
+        private const int BlurCommand = 2;
+
+        /// <summary>
+        /// The name of this view manager. This will be the name used to 
+        /// reference this view manager from JavaScript.
+        /// </summary>
+        public override string  Name
+        {
+            get
+            {
+                return "WindowsControl";
+            }
+        }
+
+        /// <summary>
+        /// The commands.
+        /// </summary>
+        public override IReadOnlyDictionary<string, object> CommandsMap
+        {
+            get
+            {
+                return new Dictionary<string, object>
+                {
+                    { "focus", FocusCommand },
+                    { "blur", BlurCommand },
+                };
+            }
+        }
+
+        /// <summary>
+        /// Exported view constants.
+        /// </summary>
+        public override IReadOnlyDictionary<string, object> ExportedViewConstants
+        {
+            get
+            {
+                return new Dictionary<string, object>
+                {
+                    { "Keys", KeyHelpers.GetKeyConstants() },
+                };
+            }
+        }
+
+        /// <summary>
+        /// Exported set of bubbling events.
+        /// </summary>
+        public override IReadOnlyDictionary<string, object> ExportedCustomBubblingEventTypeConstants
+        {
+            get
+            {
+                return new Dictionary<string, object>
+                {
+                    {
+                        "topFocus",
```

Comment:
>topFocus [](start = 25, length = 8)

We can probably put these in the "global" bubbling event types in UIManagerModule.Constants.cs so we don't have to put them in multiple places - there are less "common" events that are included there, so the bar is pretty low.

Reply:
Ok

Reply:
Ok, moved.

---
In reply to: [155706648](https://github.com/Microsoft/react-native-windows/pull/1546#discussion_r155706648) [](ancestors = 155706648)